crates/sui-framework/packages/sui-framework/sources/coin.move

```diff
@@ -556,6 +634,11 @@ public fun create_regulated_currency<T: drop>(
         coin_metadata_object: object::id(&metadata),
         deny_cap_object: object::id(&deny_cap),
     });
+    dynamic_field::add(
```

Comment:
Since we're allowing both V1 and V2 denycaps, do we really need to store the id of the deny_cap in the marker? **It can get invalid the moment you migrate V1 to V2.**

Reply:
Yeah that's fair, the migration is giant PiTA that basically makes everything potentially inconsistent except for marking the metadata as regulated/not. Will revert to that